File: Python/Log_File_Analyzer_and_Alert_System/services/error_extractor.py
import os

def extract_error_entries(entries):
    results = []
    for e in entries:
        if e._level == "ERROR":
            results.append(e)
    return results

def write_errors_to_file(error_entries, filepath):
    os.makedirs(os.path.dirname(filepath), exist_ok=True)
    with open(filepath, "w") as f:
        for e in error_entries:
            f.write(f"{e._timestamp} | {e._level} | {e._message}\n")

File: Log_File_Analyzer_Alert_System/Services/ErrorExtractor.cs
using Models;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Services
{
    public static class ErrorExtractor
    {
        public static List<LogEntry> ExtractErrorEntries(List<LogEntry> entries)
        {
            List<LogEntry> results = new List<LogEntry>();
            foreach (LogEntry entry in entries)
            {
                if (entry.Level == "ERROR")
                {
                    results.Add(entry);
                }
            }
            return results;
        }

        public static void WriteErrorsToFile(List<LogEntry> errorEntries, string filepath)
        {
            Directory.CreateDirectory(Path.GetDirectoryName(filepath));
            using (StreamWriter writer = new StreamWriter(filepath, false)) // false = overwrite
            {
                foreach (LogEntry entry in errorEntries)
                {
                    writer.WriteLine($"{entry.Timestamp} | {entry.Level} | {entry.Message}");
                }
            }
        }
    }
}
